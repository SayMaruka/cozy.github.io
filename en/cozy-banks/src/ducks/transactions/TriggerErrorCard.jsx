import React from 'react'
import { useRedirectionURL } from 'components/effects'
import { translate } from 'cozy-ui/transpiled/react'
import compose from 'lodash/flowRight'
import { withClient } from 'cozy-client'
import withBreakpoints from 'cozy-ui/transpiled/react/helpers/withBreakpoints'
import Infos from 'cozy-ui/transpiled/react/Infos'
import { ButtonLink } from 'cozy-ui/transpiled/react/Button'

const TriggerErrorCard = ({
  t,
  trigger,
  index,
  count,
  client,
  bankName,
  breakpoints
}) => {
  const url = useRedirectionURL(client, 'io.cozy.accounts', {
    account: trigger.message.account,
    konnector: trigger.message.konnector
  })
  return (
    <Infos
      className="u-bdrs-0 u-maw-none u-p-1-half"
      actionButton={
        <ButtonLink
          theme="secondary"
          extension={breakpoints.isMobile ? 'full' : 'narrow'}
          className="u-mh-0"
          label={t('Transactions.trigger-error.cta')}
          icon="openwith"
          href={url}
        />
      }
      title={
        t('Transactions.trigger-error.title') +
        (count > 1 ? ` (${index + 1}/${count})` : '')
      }
      text={t('Transactions.trigger-error.description', { bankName })}
      icon="warning"
      isImportant
    />
  )
}

export default compose(
  withClient,
  translate(),
  withBreakpoints()
)(TriggerErrorCard)
