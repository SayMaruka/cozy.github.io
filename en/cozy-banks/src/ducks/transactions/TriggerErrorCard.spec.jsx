/* global mount */
import React from 'react'
import { DumbTriggerErrorCard as TriggerErrorCard } from './TriggerErrorCard'
jest.mock('components/effects', () => ({
  useRedirectionURL: () => 'http://redirection'
}))

describe('trigger error card', () => {
  const setup = () => {
    const wrapper = mount(
      <TriggerErrorCard
        t={x => x}
        breakpoints={{ isMobile: true }}
        count={1}
        index={0}
        trigger={{
          current_state: {
            last_error: 'LOGIN_FAILED'
          },
          message: {
            account: '1234',
            konnector: 'boursorama83'
          }
        }}
      />
    )
    return {
      wrapper
    }
  }
  it('should render correctly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
