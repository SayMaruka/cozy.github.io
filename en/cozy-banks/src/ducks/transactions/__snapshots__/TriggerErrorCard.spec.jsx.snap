// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trigger error card should render correctly 1`] = `
<TriggerErrorCard
  breakpoints={
    Object {
      "isMobile": true,
    }
  }
  count={1}
  index={0}
  trigger={
    Object {
      "current_state": Object {
        "last_error": "LOGIN_FAILED",
      },
      "message": Object {
        "account": "1234",
        "konnector": "boursorama83",
      },
    }
  }
/>
`;
